spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/main/java/org/springframework/boot/configurationprocessor/metadata/ConfigurationMetadata.java

```diff
@@ -167,7 +167,8 @@ private ItemMetadata findMatchingItemMetadata(ItemMetadata metadata) {
 
 	private List<ItemMetadata> getCandidates(String name) {
 		List<ItemMetadata> candidates = this.items.get(name);
-		return (candidates != null ? new ArrayList<>(candidates) : new ArrayList<>());
+		return (candidates != null ? new ArrayList<>(candidates)
```

Comment:
I am not sure I am keen to merge this. This would return an immutable collection in one case and a mutable in the other. Thoughts?

Reply:
@snicoll I thought the same thing but I just chose this as it's a smaller change. How about inlining it in `findMatchingItemMetadata()` as follows?

```
		List<ItemMetadata> candidates = this.items.get(metadata.getName());
		if (candidates == null || candidates.isEmpty()) {
			return null;
		}
		candidates = new ArrayList<>(candidates);
```

Reply:
That works, I guess (using `ObjectUtils.isEmpty`)

Reply:
@snicoll `spring-boot-configuration-processor` doesn't depend on Spring Framework, so the above form is okay, right?

Reply:
@snicoll I just updated as the above. Please let me know if I miss something.